app/models/container_deployment.rb

```diff
@@ -6,5 +6,356 @@ class ContainerDeployment < ApplicationRecord
+            'owner_last_name'  => 'temp'
+          },
+          :tags           => nil,
+          :ws_values      => {
+            "ssh_public_key" => ssh_auth.public_key
+          }
+        },
+      }
+      parameters.merge parameters_provision
+    else
+      parameters_existing = { # non-managed + managed-existing
+        :deployment_master        => roles_addresses("deployment_master"),
+        :masters                  => roles_addresses("master"),
+        :nodes                    => roles_addresses("node"),
+        :inventory                => generate_ansible_inventory,
+        :rhel_subscribe_inventory => generate_ansible_inventory_for_subscription,
+      }
+      parameters.merge parameters_existing
+    end
+  end
+
+  def vm_fields(type, params)
+    options = {}
+    options["customization_template_id"] = CustomizationTemplate.find_by_name("SSH addition template").id
+    options["vm_name"] = type.include?("master") ? params["master_base_name"] : params["node_base_name"]
+    options["vm_memory"] = params[type + "_vm_memory"] if params[type + "_vm_memory"]
+    options["cpu"] = params[type + "_cpu"] if params[type + "_cpu"]
+    options["instance_type"] = params[type + "_instance_type"] if params[type + "_instance_type"]
+    options
+  end
+
+  def generate_ssh_keys
+    require 'sshkey'
+    keys = SSHKey.generate
+    [keys.ssh_public_key, keys.private_key]
+  end
+
+  def create_deployment_authentication(authentication)
+    auth = AUTH_CONSTS[authentication["type"]].new
+    auth.authtype = AUTHENTICATIONS_NAMES[authentication["type"]]
+    auth.assign_values authentication
+    authentications << auth
+    save!
+  end
+
+  def create_deployment_nodes(nodes)
+    work_by_vm_id = method_type.include? "existing_managed"
+    nodes.each do |node|
+      container_deployment_node = ContainerDeploymentNode.new
+      container_deployment_node.address = node["name"] unless work_by_vm_id
+      container_deployment_node.vm = Vm.find(node["id"]) if work_by_vm_id
+      container_deployment_nodes << container_deployment_node
+      node["roles"].each do |key, value|
+        if value
+          container_deployment_node.tag_add key
+        end
+      end
+      container_deployment_node.save!
+    end
+  end
+
+  # will be removed once moving to yaml format, no need to review method
+  def add_role(role_name)
+    template = <<eos
+[#{role_name}]
+#{send("#{role_name}_addresses".to_sym).join("\n")}
+eos
+    template
+  end
+
+  def create_deployment(params, user)
+    self.version = "v3"
+    self.kind = params["provider_type"].include?("openshiftOrigin") ? "origin" : "openshift-enterprise"
+    self.method_type = params["method_type"]
+    create_needed_tags
+    if method_type.include? "existing_managed"
+      self.deployed_on_ems = ExtManagementSystem.find params["underline_provider_id"]
+    elsif method_type.include? "provision"
+      ContainerDeployment.add_basic_root_template
```

Comment:
https://github.com/ManageIQ/manageiq/pull/9315 added it here, removing this line.